ksp_plugin_adapter/collision_reporter.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+
+namespace principia {
+namespace ksp_plugin_adapter {
+
+internal class CollisionReporter : UnityEngine.MonoBehaviour {
+  private void FixedUpdate() {
+    collisions.Clear();
+  }
+
+  private void OnCollisionEnter(UnityEngine.Collision collision) {
```

Comment:
I would vote for skipping the `Add` if `collision.collider` happens to be `null`.  That would simplify the client code somewhat.

Reply:
It is not immediately obvious that this would be equivalent: what if the collider dies during the physics step?
Note that Unity objects have custom comparison operators, so `!= null` checks not just that the `object` is not null (which would remain true as long as we keep a reference to it), but also that the `UnityEngine.Object` has not been destroyed.